servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -299,18 +298,20 @@ public void onComplete() {
                                     break;
                                 }
                                 if (Host.connectionsUpdater.compareAndSet(host, existing, connectionRemoved)) {
-                                    newCnx.closeAsync().subscribe();
+                                    needClose = true;
                                     break;
                                 }
                                 existing = connections;
                             }
 
-                            return failed(new IllegalStateException("LoadBalancer has closed"));
+                            final Single<C> lbClosed = failed(new IllegalStateException("LoadBalancer has closed"));
+                            return needClose ? newCnx.closeAsync().concat(lbClosed) : lbClosed;
```

Comment:
I use `needClose` to keep the existing behavior. But because connection closure is idempotent we may simplify and always close it.

Reply:
I think it is better to keep `needClose` since we are anyways doing the heavy-lifting to figure out who should close the connection anyways.